### PR TITLE
keep private files private 

### LINKED ISSUE #47: Apache docker image root folder
Should contain only the necessary website files.

### PR DESCRIPTION
**Pull request by Alexandru-Bulai**
#47  # PR 16-01-24
## Overview:
In this PR i adjusted the JS file in order to respect the lintern rules and moved the the files needed to run the website into a public folder.
## Key changes : 
1) JS file adjustments:
   - Change variables on camelCase to comply with the lintern rules
2) Public folder :
   - Move files needed for the website into a public folder for better management

### CODE REVIEW

**Comment by vladiiancu** [comment]
New rule is that any PR will be merged only if it passes all checks.

**Review by vladiiancu — CHANGES_REQUESTED** [review]
I am not a big fan of this change. 
While it seems to solve the issue one, what about in the future? When we will add new files and folders in the root project (and not only!) and they will also get copied over. 

What will you do then? Always keep this file in sync? It is prone to error. 

Can you think of another way to solve this issue?

File: JavaScript/common.js [hunk 1]
```diff
@@ -1,51 +1,51 @@
 // Drop down menu functionality
-const nav_button = document.getElementById("nav_button");
```
Comment (vladiiancu) [inline JavaScript/common.js:2]: Changing the class name is unnecessary

File: .dockerignore [hunk 2]
```diff
@@ -0,0 +1,10 @@
+# Ignore everything
+*
+
+# Whitelist method
```
Comment (vladiiancu) [inline .dockerignore]: ok, but now you just inverted the original issue. 
If you add a CSS file, what now? You still need to update this file to include it in the whitelist, right? 

Find the root cause and fix that.

**Review by vladiiancu — CHANGES_REQUESTED** [review]
Not sure why JS changes are part of this PR though. 
The rest of the comments remain

**Comment by Alexandru-Bulai** [comment]
> New rule is that any PR will be merged only if it passes all checks.

This is the only reason why i made few changes on JS file.... to pass the lintern so it can be merged when i solve the problem with the dockerignore

**Review by vladiiancu — CHANGES_REQUESTED** [review]
You are on a better path. 
You moved some files around to better organize.

My recommendation is simpler. Drop the docker ignore file as it is not needed at this point. 

Put all the files required for the website in a folder such as "public" or "web" or similar.

Adjust the docker image to only copy the contents of that folder in the apache server. 

And you are done. Easy and clean

File: README.md [hunk 3]
```diff
@@ -9,9 +9,10 @@ This project involves creating a responsive website for Pet Care company that fe
 ## Setup and Usage
 
 1. Clone the repository to your local machine.
-2. Run `docker-compose up -d` OR `docker compose up -d`.
-3. Visit http://localhost:8081/home.html
-4. You are all set up to edit the webpage! Enjoy 😄
+2. On the terminal `cd public`
+3. Run `docker-compose up -d` OR `docker compose up -d`.
```
Comment (vladiiancu) [inline README.md]: Why would I run docker from the public folder? 
Is compose.yaml a file that should be public to an end user?

File: .gitignore
Comment (vladiiancu) [inline .gitignore:1]: Why is gitignore removed?

**Review by vladiiancu — CHANGES_REQUESTED** [review]

**Review by vladiiancu — CHANGES_REQUESTED** [review]
I know this has been a lot of back and forth.
But we are on a good path. 

Now let's tidy things up, please. Adjust the PR title and description to match what you actually did :)

**Review by vladiiancu — APPROVED** [review]
I changed the pr title and we are now good to go

**Pull request merged**